onnxruntime/core/framework/session_state_initializer.cc

```diff
@@ -397,8 +403,13 @@ common::Status SaveInputOutputNamesToNodeMapping(const onnxruntime::Graph& graph
       // copy to/from CPU to go through the control flow nodes where possible/applicable.
       // the processing for the subgraph where the implicit input is consumed will do the real check on whether any
       // copy to a different device is required
-      SessionState::NodeInfo node_info(std::numeric_limits<size_t>::max(), &node, kci);
       for (const auto& input_def : node_implicit_inputs) {
+        int arg_index;
+        //Question: the implicit input may not be found in this session state name to id map, but in parent session state name to id map.
+        //@Scott
```

Comment:
@skottmckay  Please help to review this piece of codes. The code is trying to get the location from execution plan, however, I'm guessing that in this case, the execution plan / name to id map does not contain those implicit inputs, which should be in parent session state?

Reply:
The comment above talks about the current setup. Currently all control flow nodes are on CPU and we (inefficiently) make sure all their inputs (explicit and implicit) are on CPU as well. Due to that the location is implicitly known.

There's a task to avoid this copy if possible, but that will be complex and take a while to do. 

---
In reply to: [310321322](https://github.com/microsoft/onnxruntime/pull/1553#discussion_r310321322) [](ancestors = 310321322)